### ISSUE TITLE
Create About page on Flutter

### ISSUE THREAD

**Issue by MMSU-1987** [annazwicker/cat-partners#16]
Will be using Flutter to create the About page
main difference being that there will be tabs for each page to describe how to use each page
![image](https://github.com/annazwicker/cat-partners/assets/110742858/ef4f9dbf-7073-47a7-95c0-9e16be07e6f1)

**Comment by MMSU-1987** [comment]
Will start by learning about anything that seems relevant(color boxes, text position, tabs, etc.)

**Comment by MMSU-1987** [comment]
Currently figuring out how to put a title for the About page, much like how it is on Figma. Thinking about using Container.

**Comment by MMSU-1987** [comment]
Currently have vague idea about how to format About page

Using both Row and Columns to align the boxes together and to keep the title and boxes separate from each other.

**Comment by MMSU-1987** [comment]
Stopping for now.
Current progress:
- Got idea of using rows within columns widget to make the 2 boxes
- issue with making boxes due to supposed lack of constraints(i thought i added those already)
- For everyone else, for title formatting, use this -> Text('About This Site', style: TextStyle(fontWeight: FontWeight.bold, height: 2, fontSize: 30 )). This is to make sure it is uniformed. Will need to figure out font later(playful display regular)
- worked for 2 hrs

**Comment by MMSU-1987** [comment]
Began working on solving box issue. SizedBox might be a good option.
https://stackoverflow.com/questions/48857858/how-to-create-colour-box-with-fixed-width-and-height-in-flutter

**Comment by MMSU-1987** [comment]
Getting closed to boxes and text positions. Had to use a Stack(allows for widgets to be placed a top one another).
Current progress showen below
![image](https://github.com/annazwicker/cat-partners/assets/110742858/57357c6d-095b-4384-bbc0-764319f202c8)

**Comment by MMSU-1987** [comment]
Stopping work for now.
Current progress
- Figured out how to properly align text and box
- Will need to add link to Cat Partners SU page
- Need to figure out Widget for scrolling through cat pics(and likely adding new ones
- Worked 1hr
- Total: ~5 hrs (meetings + individual time)

![image](https://github.com/annazwicker/cat-partners/assets/110742858/ebe761fb-2457-40fe-8902-09a67d7ceb0c)

**Comment by MMSU-1987** [comment]
Forgot to make issue when starting work.
Worked for an 1hr(~7pm to 8pm)
Worked on adjusting code to handle various screen sizes
Would now like to work on the following:
- Add placeholders for our images
- Add widget to show cat images(and add functionality to change them)
- Add how-to-use-site section with tabs
- Adjust code to make it look more readable

Images below show how layout changes with screen size
![image](https://github.com/annazwicker/cat-partners/assets/110742858/64ad00f3-2ac7-4d79-904c-c53654dbb0f2)
![image](https://github.com/annazwicker/cat-partners/assets/110742858/c44938a6-70d9-416c-8b7d-7267c28a6d54)

**Comment by MMSU-1987** [comment]
Started work. Will try to work on adding a horizontal scroll for tabs.

**Comment by MMSU-1987** [comment]
Added vertical scroll bar. Will need to add box for tabs to add horizontal scroll.
![image](https://github.com/annazwicker/cat-partners/assets/110742858/5c7d9682-deb2-4a8f-a1dd-1d426cde28b0)

**Comment by MMSU-1987** [comment]
Made some errors with that affected layout, need to reverse changes

Forgot to mention previously, but did some encapsulation.

**Comment by MMSU-1987** [comment]
Errors with alignment(everything is aligning left instead of center, including scroll bars). Currently figuring out issue.

**Comment by MMSU-1987** [comment]
Stopping work for now. Fixed issue of alignment, needed to move the position of Center to where it encompasses the majority of the code. Also did some encapsulation to make the main code look less big and less cluttered. Made mock-up to simulate where tabs will go. Bigger Issue: had trouble fetching Daniel's changes to the pages bar using pull/fetch/etc. Tried simply copying and pasting his work, but it seemed to cause issues to where the entire shows errors. Will need to notify Daniel about.
Once fixed, will work on:
- Tabs layout
- Tabs content

Worked for ~2 hrs

**Comment by MMSU-1987** [comment]
Started work. Not 100% sure what to work on. Will probably see how putting images work.

**Comment by MMSU-1987** [comment]
First attempt to add custom image, didn't work, will attempt to fix.

**Comment by MMSU-1987** [comment]
Figured out method to add custom images(as seen below). Ignore image used, chosen for testing purposes. Need to figure out scrollable set of cat images.
![image](https://github.com/annazwicker/cat-partners/assets/110742858/a71507ce-0c56-43ee-b05e-6a3f97dccb6c)
(unrelated, but i chuckle every time I see the test image)

**Comment by MMSU-1987** [comment]
Stopping work for now. Have figured out how to get images. However, 2nd set of test images not working properly, will need to figure that out. Daniel mentioned that those images for the cats need to be managed by the admins(or at the very least, changed by someone) when the time comes. Will need to look into that. Also need to tell Daniel about the issue mentioned in the last work session.
![image](https://github.com/annazwicker/cat-partners/assets/110742858/025eee74-4ff9-4478-a475-26589c9e24c9)

**Comment by MMSU-1987** [comment]
Started work. Plan to making small adjustments and add tabs bar for How-To-Use-Pages.

**Comment by MMSU-1987** [comment]
Got a set of tabs working. Need to center them and add contents to them. 
![image](https://github.com/annazwicker/cat-partners/assets/110742858/dd0b105f-2126-4885-b874-c43dfc4cd939)

**Comment by MMSU-1987** [comment]
Fixed tab issues and set up tabviews.
Will need to do add the contents for each of the tabs.
Other general things i need to do
- figure out fonts
- make adjustments based on window size
- color scheme

![image](https://github.com/annazwicker/cat-partners/assets/110742858/bd496a49-940d-4bc8-a423-a324d4207519)

**Comment by MMSU-1987** [comment]
Started work rn. Plan to at least get font ready.

**Comment by MMSU-1987** [comment]
Added fonts to the page. Will need to inform the others about the new font and push it for their use. Now working on way to adjust widgets with screen size.
![image](https://github.com/annazwicker/cat-partners/assets/110742858/b5ba3d7d-782a-4ff8-a194-1088d1ec43e2)

**Comment by MMSU-1987** [comment]
Made how-to section size adjustable with screen size. Will likely need to discuss screen size with group. Now need to remove horizontal scroll.
![image](https://github.com/annazwicker/cat-partners/assets/110742858/20b9b143-31e4-48d5-8fe7-2046e7cf67c0)

**Comment by MMSU-1987** [comment]
Stopping work for now, worked for ~2 hrs
Have begun adding the contents for the how-to-use tabs. A lot of the contents can't be added now due to not having full idea about how pages will work.
The page is almost finished. The main thing missing are the how-to contents, the images of us and the cats, and adding the function for admins to add and remove images. The rest is more small refinements and adjustments.
![image](https://github.com/annazwicker/cat-partners/assets/110742858/e5f3aef8-4358-4276-9804-c6cd656effee)

**Comment by MMSU-1987** [comment]
Started research and work on how to let users(specifically admins) add new cat images to mini cat gallery
![image](https://github.com/annazwicker/cat-partners/assets/110742858/9ae4c46a-a0e8-476b-b5e8-dfec2f625b72)

**Comment by MMSU-1987** [comment]
found a variety of sources:
https://stackoverflow.com/questions/44841729/how-to-upload-images-to-server-in-flutter 
https://cloudinary.com/documentation/upload_images_in_flutter_tutorial
https://firebase.google.com/docs/storage/flutter/upload-files

Main issue is that requires could storage stuff. Not knowledgeable enough to mess around with it, so will ask the others about it.

Also adding the contents for the Accounts tab

**Comment by MMSU-1987** [comment]
Stopped work(worked for an hr). Mostly cause most things required questions from team members.

Will need to figure out how to:
- let users upload images
- how to let certain users have permission to upload

Worked total 4hr 30 mins this week

**Comment by MMSU-1987** [comment]
Started work. Plan on continue work tasked on Tuesday and started this morning.

**Comment by MMSU-1987** [comment]
Made appropriate adjustments to about page based on group advice. Now would need to learn how to incorporate database into site, so we can allow admins to upload and remove images
![image](https://github.com/annazwicker/cat-partners/assets/110742858/b4a0ef6f-fcd3-4ff0-8d15-d4ad65922b55)
![image](https://github.com/annazwicker/cat-partners/assets/110742858/a918f21b-670e-4a54-81e5-c771da9b92b4)

**Comment by MMSU-1987** [comment]
Stopped work. Worked for about an hr.

After making those major adjustments, made slight modifications, made comments, etc.
Currently don't like how vertical cat galley looks like. Will change it so it is 2 per row. Will then need to figure how that will be done when users get to upload and remove images.

**Comment by MMSU-1987** [comment]
Started work. Will work on Tab contents for Account page(based off of what I pulled from main) and research interaction with google cloud to let admins remove and upload images.

**Comment by MMSU-1987** [comment]
Got general content for Account page. Just need to change formatting and remove potential overflow errors.

**Comment by MMSU-1987** [comment]
Still making changes and adjustments to content to make it look better

**Comment by MMSU-1987** [comment]
Stopping work for now. Worked for 1 hr 30min
Didnt get around to attempting to implement google cloud interaction(Daniel even said google API would work so not 100% sure there). Will look into it next time.

Got more reliable format for text, just need to readjust box.

**Comment by MMSU-1987** [comment]
Started work. Continue working on Tab contents, make small adjustments and research google cloud interaction.(hopefully)

**Comment by MMSU-1987** [comment]
Made necessary format changes to tab section and feel good about current contents of Account tab. Home, Admin, and Sign Up to Feed pages are still not complete so will wait on those. Will look into google cloud(or Google API as Daniel suggested at one point)

**Comment by MMSU-1987** [comment]
Some links here that I (or maybe we) will likely need or find useful
Get all contents of a bucket(where images will be stored): https://cloud.google.com/storage/docs/samples/storage-list-files#storage_list_files-java 
Upload images from own files and camera: https://medium.com/geekculture/flutter-how-to-upload-photos-taken-from-the-camera-and-other-files-via-http-386d04218e02
Uploading files and photos with Firebase dependencies: https://youtu.be/HCmAwk2fnZc?si=9Dpxp2Q1CqiVCfxx   (idk about this one)

**Comment by MMSU-1987** [comment]
Stopping work for today. Worked for 1hr 30min.
Main issue comes with how to interact with google cloud database. Plus, many sites mention using Firebase. Jayden said that Firebase is document-based(JSON) while Google Cloud is relational. Using it might cause conflict. Will need to find more sites that use google cloud for better reference.
Firebase: how to differentiate admins: https://stackoverflow.com/questions/66344532/how-can-i-set-admin-button-for-admins-in-flutter

Work total: ~5 hr this week

**Comment by MMSU-1987** [comment]
Started Work. Since there is not much for me to do, I will attempt to add in the Google Cloud interaction with buckets to upload and remove images from mini cat gallery.

**Comment by MMSU-1987** [comment]
Had issues with dependencies. After what happened with main, scared to change dependencies around. Will discuss with group about it. Will instead look at videos and stuff and make comments in code for future work on this topic.

**Comment by MMSU-1987** [comment]
Stopped work for now. Worked for ~hr.
Looked at videos and attempted to understand them(Still confused on some aspects, especially since video had different end goal shown, different from what i would need).
https://www.youtube.com/watch?v=CV0vGWDtCR0
https://hrishikeshpathak.com/blog/file-picker-flutter/  -- potentially use for file picking instead of camera using

Will need to figure out what else I can do.

**Comment by MMSU-1987** [comment]
Started Work. Will officially attempt to add buttons to upload and remove images from mini cat gallery

**Comment by MMSU-1987** [comment]
Got buttons added. The one to add images does bring up folder, so that works, but other button is tasked with removing it from storage(from bucket). Need to look into it.

**Comment by MMSU-1987** [comment]
Stopped work for now. Worked for ~hr. Found it difficult add delete function since it requires the url, which with the current setup is difficult. Idea to allow admins to press an "X" button to remove image. Each instance of a cat image will store its url from the bucket. Will need to discuss with Daniel about it since he is more knowledge able about Google Cloud right now.

Otherwise, made minor progress despite that. I feel like it is unlikely for me to get the 5hrs this week. Need input from Daniel to continue.

**Comment by MMSU-1987** [comment]
After working on Google sign in page, made adjustments to How-to-use about page. Gonna fill in the other tabs as well.

**Comment by MMSU-1987** [comment]
Stopped work, worked for 1hr and 30mins.

Mainly did formatting and revisions.
Will likely be meeting with group later to work on presentation and to help with formatting and layout.
Will work on more page revisions next time.

**Comment by MMSU-1987** [comment]
Started Work. Need to make adjustments based on new text in How-To-Use tabs(box size, more text, etc.)

**Comment by MMSU-1987** [comment]
Almost finished all tabs in the About Page. Also change sized of its box to make less empty space. Will now fix top box and hard code in cat images for presentation.

**Comment by MMSU-1987** [comment]
Put the cat images onto the About page. They look funny.
Will now see if there are any changes I can make to the Account page
![image](https://github.com/annazwicker/cat-partners/assets/110742858/5e39b634-8d00-46bb-8800-ee4489e90925)

**Comment by MMSU-1987** [comment]
Stopped work. Worked for 1hr 30min.
Made Google sign in, About, and Account page more presentable. Will discuss with others about any needed changes.
Will attempt to add backend next time after putting final touches.

Worked total of 6 hrs this week

**Comment by annazwicker** [comment]
Making various changes to the about page.  Mostly grammatical changes or rephrasing sentences to be clearer.

**Comment by annazwicker** [comment]
Committed changes to restored-commit.  Also made a quick change to the readme.  Stopping work (a little more than 3.5 hours).
